inst/get-testdata/score_untrained-mean.R

```diff
@@ -0,0 +1,187 @@
+  select(model_id, importance) |>
+  mutate(
+    ens_mthd = "simple_ensemble-mean",
+    algorithm = "lomo",
+    test_purp = "properly assigned"
+  ) |>
+  as.data.frame()
+
+## Case 3: no missing data and 'simple_ensemble' using agg_fun = median
+# ensemble built from all models
+ens_df_simple <- simple_ensemble(
+  dat_mean,
+  model_id = "ens_all", agg_fun = "median"
+)
+# construct ensembles without each model
+model_names <- dat_mean$model_id
+for (i in seq_along(model_names)) {
+  sub_data <- dat_mean[model_names != model_names[i], ]
+  sub_ens <- simple_ensemble(sub_data,
+    model_id = paste0("ens_wo_", model_names[i]),
+    agg_fun = "median"
+  )
+  ens_df_simple <- bind_rows(ens_df_simple, sub_ens)
+}
+# evaluate each ensemble
+ens_df_simple <- score_model_out(ens_df_simple,
+  target_data_mean,
+  metrics = "se_point"
+) |>
+  # calculate importance scores: subtract the error of the ensemble-all
+  mutate(importance = se_point - se_point[1]) |>
+  filter(model_id != "ens_all")
+ens_df_simple$model_id <- sub("^ens_wo_", "", ens_df_simple$model_id)
+# expected importance scores with mean output and simple median ensemble
+exp_imp_mean3 <- ens_df_simple |>
+  select(model_id, importance) |>
+  mutate(
+    ens_mthd = "simple_ensemble-median",
+    algorithm = "lomo",
+    test_purp = "properly assigned"
+  ) |>
+  as.data.frame()
+
+
+## Case 4: a missing data and 'simple_mean' ensemble
+# data with missing values
+sub_dat_mean <- dat_mean |> filter(model_id %in% model_id_list[c(1, 3)])
+# ensemble built from all models
+ens_df <- simple_ensemble(sub_dat_mean, model_id = "ens_all")
+# construct ensembles without each model
+model_names <- sub_dat_mean$model_id
+for (i in seq_along(model_names)) {
+  sub_data <- sub_dat_mean[model_names != model_names[i], ]
+  sub_ens <- simple_ensemble(sub_data,
+    model_id = paste0("ens_wo_", model_names[i])
+  )
+  ens_df <- bind_rows(ens_df, sub_ens)
+}
+# evaluate each ensemble
+ens_df <- score_model_out(ens_df,
+  target_data_mean,
+  metrics = "se_point"
+) |>
+  # calculate importance scores: subtract the error of the ensemble-all
+  mutate(importance = se_point - se_point[1]) |>
+  filter(model_id != "ens_all")
+ens_df$model_id <- sub("^ens_wo_", "", ens_df$model_id)
+# expected importance scores with mean output and simple mean ensemble
+exp_imp_mean4 <- data.frame(model_id = model_id_list) |>
+  left_join(ens_df, by = "model_id") |>
+  select(model_id, importance) |>
+  mutate(
+    ens_mthd = "simple_mean",
+    algorithm = "lomo",
+    test_purp = "missing data"
+  ) |>
+  as.data.frame()
+
+# combine the expected importance scores
+exp_imp_mean <- rbind(
+  exp_imp_mean1, exp_imp_mean2, exp_imp_mean3, exp_imp_mean4
+)
+
+# save data
+saveRDS(dat_mean,
+  file = "tests/testthat/testdata/dat_mean.rds"
+)
+saveRDS(exp_imp_mean,
+  file = "tests/testthat/testdata/exp_imp_mean_untrained_lomo.rds"
+)
+saveRDS(target_data_mean,
+  file = "tests/testthat/testdata/target_mean.rds"
+)
```

Comment:
This is a large dataset (over 200,000 rows). But only a small portion of it is used here. To reduce the size of the package, you could consider only saving the rows you need here, right?

Reply:
that said, the file sizes are small (only ~60kb)